src/main/java/com/hivemq/client/internal/mqtt/MqttProxyConfigImplBuilder.java

```diff
@@ -94,27 +98,34 @@ private void setProxyHost(final @NotNull Object host) {
         return self();
     }
 
-    public @NotNull B proxyUsername(final @Nullable String username) {
+    public @NotNull B username(final @Nullable String username) {
         this.username = username;
         return self();
     }
 
-    public @NotNull B proxyPassword(final @Nullable String password) {
+    public @NotNull B password(final @Nullable String password) {
         this.password = password;
         return self();
     }
 
-    private @NotNull InetSocketAddress getProxyAddress() {
+    public @NotNull B handshakeTimeout(final long timeout, final @Nullable TimeUnit timeUnit) {
```

Comment:
timeUnit is annotated Nullable but checked for not null.

Reply:
This is intended, the public interfaces are annotated NotNull, but the internal implementations are annotated Nullable so static code analysis warns when the not null check is missing